tests/transforms/test_defer_measurements.py

```diff
@@ -36,8 +108,8 @@ def test_only_mcm(self):
         def qnode1():
             return qml.expval(qml.PauliZ(0))
 
-        @qml.defer_measurements
         @qml.qnode(dev)
+        @qml.defer_measurements
```

Comment:
was this swap necessary?

Reply:
Yeah, because putting above the qnode adds the transform to the transform program, which doesn't transform the qnode tape until `qml.execute`. Transforming the qfunc makes the qnode tape be deferred.

Reply:
I see that it's the other order in certain tests. just wondering if the order will ever be a problem for users. seems like yes? do we need to communicate that?

Reply:
Nope, it won't be. This was just needed because I needed to inspect `qnode.tape`. When the decorator is applied to a qnode, the transform is applied by device preprocessing so it's a non-issue as that can handle any arbitrary number of tapes.